homeassistant/components/surepetcare/binary_sensor.py

```diff
@@ -44,65 +63,77 @@ class SurePetcareBinarySensor(BinarySensorDevice):
     """A binary sensor implementation for Sure Petcare Entities."""
 
     def __init__(
-        self, _id: int, name: str, spc, device_class: str, sure_type: SureThingID
+        self: BinarySensorDevice,
+        _id: int,
+        spc: SurePetcareAPI,
+        device_class: str,
+        sure_type: SureProductID,
     ):
         """Initialize a Sure Petcare binary sensor."""
         self._id = _id
-        self._name = name
-        self._spc = spc
-        self._device_class = device_class
         self._sure_type = sure_type
-        self._state = {}
+        self._device_class = device_class
+
+        self._spc: SurePetcareAPI = spc
+        self._spc_data: Dict[str, Any] = self._spc.states[self._sure_type].get(self._id)
+        self._state: Dict[str, Any] = {}
+
+        _LOGGER.debug("%s -> self._spc_data: %s", self._id, self._spc_data)
+
+        # cover special case where a device has no name set
+        if "name" in self._spc_data:
+            name = self._spc_data["name"]
+        else:
+            name = f"Unnamed {self._sure_type.name.capitalize()}"
+
+        self._name = f"{self._sure_type.name.capitalize()} {name.capitalize()}"
 
         self._async_unsub_dispatcher_connect = None
 
     @property
-    def is_on(self):
+    def is_on(self) -> Optional[bool]:
         """Return true if entity is on/unlocked."""
         return bool(self._state)
 
     @property
-    def should_poll(self):
+    def should_poll(self) -> bool:
         """Return true."""
         return False
 
     @property
-    def name(self):
+    def name(self) -> str:
         """Return the name of the device if any."""
         return self._name
 
     @property
-    def device_state_attributes(self):
-        """Return the state attributes of the device."""
-        return self._state
-
-    @property
-    def device_class(self):
+    def device_class(self) -> str:
         """Return the device class."""
-        return DEFAULT_DEVICE_CLASS if not self._device_class else self._device_class
+        return None if not self._device_class else self._device_class
 
     @property
-    def unique_id(self):
+    def unique_id(self: BinarySensorDevice) -> str:
         """Return an unique ID."""
-        return f"{self._spc.household_id}-{self._id}"
+        return f"spc-{self._spc_data['household_id']}-{self._id}"
```

Comment:
Changing the unique_id is a breaking change that we should avoid. We don't need any words anyway in the id. Ids are enough.

We do need a type id per entity type if a platform adds more than one entity type per device.

Reply:
fixed :)